File: BEmu/MarketDataRequest/MarketElementTime.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="MarketDataRequest/MarketElementTime.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.MarketDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class MarketElementTime : Element
    {
        private readonly Datetime _time;
        private readonly string _name;

        internal MarketElementTime(string name, DateTime time)
        {
            this._name = name;
            //this._time = new Datetime(time, Datetime.DateTimeTypeEnum.time);
            this._time = new Datetime(time);
        }

        public override Name Name { get { return new Name(this._name); } }
        public override int NumValues { get { return 1; } }
        public override int NumElements { get { return 0; } }
        public override bool IsNull { get { return false; } }
        public override Schema.Datatype Datatype { get { return Schema.Datatype.TIME; } }

        public override object this[int index]
        {
            get
            {
                if (index == 0)
                    return this._time;
                else
                    throw new IndexOutOfRangeException();
            }
        }

        internal override StringBuilder PrettyPrint(int tabIndent, bool surroundValueWithQuotes = false)
        {
            return base.PrettyPrintHelper(tabIndent, Types.DisplayFormats.MarketDataRequests.FormatDateZone(this._time));
        }

        public override object GetValue()
        {
            return this._time;
        }

        public override Datetime GetValueAsTime()
        {
            return this._time;
        }
    }
}

File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementTime.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/MarketElementTime.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Datetime;
import com.bloomberglp.blpapi.DateTimeTypeEnum;
import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Schema;

public class MarketElementTime extends Element
{
	private final Datetime _date;
	private final String _name;
	
	MarketElementTime(String name, Datetime date)
    {
		this._date = new Datetime(date, DateTimeTypeEnum.time);		
        this._name = name;
    }
	
    public Name name() throws Exception
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.TIME;
    }
    
    protected StringBuilder prettyPrint(int tabIndent) throws Exception
    {
    	String strDate = com.bloomberglp.blpapi.DisplayFormats.MarketDataRequests_FormatDateZone(this._date);
    	return super.prettyPrintHelper(tabIndent, strDate);
    }
    
    public Object getValue()
    {
    	return this._date;
    }

    public Datetime getValueAsTime()
    {
        return this._date;
    }
}
